src/osd/OSD.cc

```diff
@@ -8075,17 +8082,21 @@ void OSD::handle_osd_map(MOSDMap *m)
 
       OSDMap *o = new OSDMap;
       if (e > 1) {
-	bufferlist obl;
-        bool got = get_map_bl(e - 1, obl);
-	if (!got) {
-	  auto p = added_maps_bl.find(e - 1);
-	  ceph_assert(p != added_maps_bl.end());
-	  obl = p->second;
-	}
-	o->decode(obl);
+        OSDMapRef prev;
+        auto p = added_maps.find(e - 1);
+        if (p != added_maps.end()) {
+          prev = p->second;
+        } else {
+          prev = get_map(e - 1);
```

Comment:
It this case actually possible?

Reply:
```
this line will run in the case:
firt epoch osdmap is inc map, 
it will look up prev epoch osdmap,
added_maps not have the prev epoch osdmap, not found
then it will run the get_map to get prev epoch full osdmap 
```


https://github.com/ceph/ceph/blob/main/src/osd/OSD.h#L1725
```
  // osd map cache (past osd maps)
  OSDMapRef get_map(epoch_t e) {
    return service.get_map(e);
  }
```
https://github.com/ceph/ceph/blob/main/src/osd/OSD.h#L696
```
  OSDMapRef get_map(epoch_t e) {
    OSDMapRef ret(try_get_map(e));
    ceph_assert(ret);
    return ret;
  }
```

Reply:
Oh, I misread the removed code, it's checking get_map_bl first.  Ok, makes sense to me.